src/openrct2/rct1/S4Importer.cpp

```diff
@@ -594,24 +594,30 @@ class S4Importer : public IS4Importer
             dst->station_depart[i] = src->station_light[i];
 
             // Use src->station_depart[i] when we import with guests and vehicles intact
-            dst->train_at_station[i] = 0xFF;
+            dst->train_at_station[i] = src->station_depart[i];
 
             dst->entrances[i] = src->entrance[i];
             dst->exits[i] = src->exit[i];
             dst->queue_time[i] = src->queue_time[i];
-            dst->last_peep_in_queue[i] = SPRITE_INDEX_NULL;
+            dst->last_peep_in_queue[i] = src->last_peep_in_queue[i];
             dst->queue_length[i] = src->num_peeps_in_queue[i];
         }
         dst->num_stations = src->num_stations;
 
-        for (int i = 0; i < 32; i++)
+        // Vehicle links (indexes converted later)
+        for (int i = 0; i < RCT1_MAX_VEHICLES_PER_RIDE; i++)
+        {
+            dst->vehicles[i] = src->vehicles[i];
```

Comment:
Could be wrong here but shouldn't only the train vehicles (i.e. first car) end up in the ride vehicle list.